src/base/install/getFilesInDir.m

```diff
@@ -0,0 +1,140 @@
+%                                    directory. (Default: true)
+% 
+% EXAMPLES:
+%    Get all m files in the source folder:
+%    files = getFilesInDir('dirToList', [CBTDIR filesep 'src'], 'restrictToPattern', '\.m$');
+%    Get the git tracked files in the test Directory.
+%    files = getFilesInDir('dirToList', [CBTDIR filesep 'test'], 'gitTypeFlag', 'tracked');
+%    Get all git tracked files  which start with "MyFile" in the current directory
+%    files = getFilesInDir('gitTypeFlag', 'tracked', 'restrictToPattern', '^MyFile');
+%    Get only the gitIgnored files in the current folder
+%    files = getFilesInDir('gitTypeFlag', 'ignored');
+
+persistent COBRAIgnored
+
+if isempty(COBRAIgnored)
+    COBRAIgnored = regexptranslate('wildcard',getIgnoredFiles());
+end
+
+gitFileTypes = {'tracked','all','untracked','ignored','COBRAIgnored'};
+parser = inputParser();
+parser.addParamValue('dirToList',pwd,@(x) exist(x,'file') == 7);
+parser.addParamValue('gitTypeFlag','all',@(x) ischar(x) && any(strcmpi(x,gitFileTypes)));
+parser.addParamValue('restrictToPattern','',@(x) ischar(x));
+parser.addParamValue('checkSubFolders','',@(x) islogical(x) || (isnumeric(x) && x == 0 || x == 1));
+
+parser.parse(varargin{:});
+
+%get the absolute path of the files that are listed
+
+currentDir = cd(parser.Results.dirToList);
+absPath = pwd;
+%get all files in the directory.
+gitType = lower(parser.Results.gitTypeFlag);
+
+%test, whether the folder is git controlled
+[gitStatus,~] = system('git status');
+if gitStatus ~= 0 && ~strcmpi(gitType,'cobraignored')
+    gitType = 'all';
+end
+if gitStatus == 0 && strcmpi(gitType,'cobraignored')
+    [~,repos] = system('git remote -v');
+    if any(cellfun(@(x) ~isempty(strfind(x,'opencobra/cobratoolbox.git')),strsplit(repos,'\n')))
+        %So, we are on a COBRA repo. lets just use the ignored option
+        gitType = 'ignored';
+    end
+end
+
+switch gitType
+    case 'all'
+        if gitStatus == 0
+            [status, trackedfiles] = system('git ls-files');             
+            [status, untrackedfiles] = system('git ls-files -o');        
+            trackedfiles = strsplit(strtrim(trackedfiles), '\n');
+            untrackedfiles = strsplit(strtrim(untrackedfiles), '\n');
+            files = [trackedfiles,untrackedfiles];            
+        else
+           if parser.Results.checkSubFolders
+               rdircall = ['**' filesep '*'];
+           else
+               rdircall = ['*']
+           end
+           files = rdir(rdircall); 
+           files = {files.name}'; %Need to transpose to give consistent results.
+        end
+            
+    case 'tracked'
+        [status, files] = system('git ls-files');
+        files = strsplit(strtrim(files), '\n');
+    case 'untracked'
+        %Files, which are not tracked but are not ignored.
+        [status, files] = system('git ls-files -o -exclude-standard');
+        files = strsplit(strtrim(files), '\n');
+    case 'ignored'
```

Comment:
What is the difference between `'ignored'` and `'cobraignored'`?

Reply:
Cobraignored will read the COBRA ignored files file, while just ignored relies on the .gitignore of whichever project this is in, (and if the folder is not under git control, it doesn't work).

Reply:
Still not sure I understand ... Isn't this the same for this repo? Or is there a plan to use this function in another repo?

Reply:
Yes and no...
If you run this function in e.g. tutorials, he tutorials .gitignore would be used if you run it with `'ignore'`. 
Also `'ignore'`, is much fast than `'cobraignore'`, since the git functions are much faster than manually parsing the gitignore file and regexp checking each file...
However 'ignore' only works if you are in a git controlled folder AND it will use whichever .gitignore is valid for that folder.
So if you call e.g. changeCobraSolver from some non cobra-repository location, it would use the "wrong" or no .gitignore file. We could (of course) add cd's to changeCobraSolver, but I'm not sure, whether we should do that...
Thats why I implemented the two options. For `changecobraSolver` we need the explicit cobraIgnore option. for initCobraToolbox and testAll we can work with the faster git option (as we cd).

Reply:
May you add such a note to the header?